cmd/dra-example-kubeletplugin/cdi.go

```diff
@@ -89,18 +91,37 @@ func (cdi *CDIHandler) CreateClaimSpecFile(claimUID string, devices PreparedDevi
 		Devices: []cdispec.Device{},
 	}
 
-	gpuIdx := 0
-	for _, device := range devices {
+	for i, device := range devices {
+		envs := []string{
+			fmt.Sprintf("GPU_DEVICE_%d=%s", i, device.DeviceName),
+		}
+
+		if device.Config.Sharing.IsTimeSlicing() {
+			tsconfig, err := device.Config.Sharing.GetTimeSlicingConfig()
+			if err != nil {
+				return fmt.Errorf("unable to get time slicing config for device %v: %v", device.DeviceName, err)
+			}
+			envs = append(envs, fmt.Sprintf("GPU_SHARING_STRATEGY=%s", configapi.TimeSlicingStrategy))
+			envs = append(envs, fmt.Sprintf("GPU_TIMESLICE_DURATION=%v", tsconfig.Duration))
+		}
+
+		if device.Config.Sharing.IsSpacePartitioning() {
```

Comment:
nit: Can timeslicing and space partitioning be specified at the same time?

Reply:
Not in the way I put together this example driver. One must pick a strategy (either TimeSlicing or SpacePartitioning) and depending on what they picked, only one of these will be valid.

Reply:
Does an `if -- else if` (or even a `switch` block) make more sense then to make the intent clearer?

Reply:
Sure. I just really dislike if/else blocks (for aesthetic reasons) which is why I didn't do it.

Reply:
I just find this really hard to read:
```
        if device.Config.Sharing.IsTimeSlicing() {
            tsconfig, err := device.Config.Sharing.GetTimeSlicingConfig()
            if err != nil {
                return fmt.Errorf("unable to get time slicing config for device %v: %v", device.DeviceName, err)
            }
            envs = append(envs, fmt.Sprintf("GPU_SHARING_STRATEGY=%s", configapi.TimeSlicingStrategy))
            envs = append(envs, fmt.Sprintf("GPU_TIMESLICE_DURATION=%v", tsconfig.Duration))
        } else if device.Config.Sharing.IsSpacePartitioning() {
            spconfig, err := device.Config.Sharing.GetSpacePartitioningConfig()
            if err != nil {
                return fmt.Errorf("unable to get space partitioning config for device %v: %v", device.DeviceName, err)
            }
            envs = append(envs, fmt.Sprintf("GPU_SHARING_STRATEGY=%s", configapi.SpacePartitioningStrategy))
            envs = append(envs, fmt.Sprintf("GPU_PARTITION_COUNT=%v", spconfig.PartitionCount))
        }
```

How could this be done as a switch? Maybe that will look better.

Reply:
```golang
		switch {
		case device.Config.Sharing.IsTimeSlicing():
			tsconfig, err := device.Config.Sharing.GetTimeSlicingConfig()
			if err != nil {
				return fmt.Errorf("unable to get time slicing config for device %v: %v", device.DeviceName, err)
			}
			envs = append(envs, fmt.Sprintf("GPU_SHARING_STRATEGY=%s", configapi.TimeSlicingStrategy))
			envs = append(envs, fmt.Sprintf("GPU_TIMESLICE_DURATION=%v", tsconfig.Duration))
		case device.Config.Sharing.IsSpacePartitioning():
			spconfig, err := device.Config.Sharing.GetSpacePartitioningConfig()
			if err != nil {
				return fmt.Errorf("unable to get space partitioning config for device %v: %v", device.DeviceName, err)
			}
			envs = append(envs, fmt.Sprintf("GPU_SHARING_STRATEGY=%s", configapi.SpacePartitioningStrategy))
			envs = append(envs, fmt.Sprintf("GPU_PARTITION_COUNT=%v", spconfig.PartitionCount))
		}
```

Reply:
Updated.